engine/rowset/src/main/java/io/deephaven/engine/rowset/WritableRowSet.java

```diff
@@ -127,6 +127,15 @@ default WritableRowSet extract(@NotNull final RowSet other) {
      */
     void compact();
 
+    /**
+     * Reset this RowSet to match another RowSet by updating the {@code innerSet}. This will internally make a
+     * copy-on-write call of the incoming RowSet so the two sets will diverge when either is modified. To maintain
+     * continuity between the sets, this function should be called each UGP cycle.
+     *
+     * @param other the RowSet to accept as the new innerSet.
+     */
```

Comment:
This is putting too many implementation details into the interface.
```suggestion
    /**
     * Reset this RowSet to exactly match another RowSet. Subsequent modifications to {@code other} will not
     * change {@code this}.
     *
     * @param other The RowSet to reset to 
     */
```